src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -572,36 +572,61 @@ private static byte[] getJ0(byte[] iv, byte[] subkeyH, int blockSize) {
         return j0;
     }
 
-    // Wrapper function around AES-GCM interleaved intrinsic that splits
-    // large chunks of data into 1MB sized chunks. This is to place
-    // an upper limit on the number of blocks encrypted in the intrinsic.
+    /**
+     * Wrapper method around Combined AES-GCM intrinsic that splits
+     * large chunks of data into 1MB sized chunks. This is to place
+     * an upper limit on the number of blocks encrypted in the intrinsic.
+     *
+     * For decrypting in-place byte[], calling methods must ct must set to null
+     * to avoid combined intrinsic, call GHASH directly before GCTR to avoid
+     * a bad tag exception. This check is not performed here because it would
+     * impose a check every operation which is less efficient.
+     *
+     * The value of ctOfs is irrelevant if ct is null.
+     */
     private static int implGCMCrypt(byte[] in, int inOfs, int inLen, byte[] ct,
                                     int ctOfs, byte[] out, int outOfs,
                                     GCTR gctr, GHASH ghash) {
 
         int len = 0;
-        if (inLen > SPLIT_LEN) {
+        // Loop if input length is greater than the SPLIT_LEN
+        if (inLen > SPLIT_LEN && ct != null) {
+            int partlen;
             while (inLen >= SPLIT_LEN) {
-                int partlen = implGCMCrypt0(in, inOfs + len, SPLIT_LEN, ct,
+                partlen = implGCMCrypt0(in, inOfs + len, SPLIT_LEN, ct,
```

Comment:
why not `int partlen` and get rid of line 594

Reply:
can reuse the same partlen for all loops through the while